src/main/java/core/basesyntax/TreeSetCharacters.java

```diff
@@ -17,7 +23,28 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        Set<Character> uniqueChars = new TreeSet<>();
+        try (FileReader fileReader = new FileReader(fileName)) {
+            int data = fileReader.read();
+            while (data != -1) {
+                if (Character.isLetter(data)) {
+                    uniqueChars.add((char) Character.toLowerCase(data));
+                }
+                data = fileReader.read();
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            throw e;
+        }
+        StringBuilder result = new StringBuilder();
+        int n = 0;
+        for (char c : uniqueChars) {
+            result.append(c);
+            if (++n == 5) {
```

Comment:
can we use result.length() for checking?

Reply:
no, simplified